doozerlib/dblib.py

```diff
@@ -1,64 +1,366 @@
+            existing_columns = set(self._table_column_cache[table_name])
+
+        missing_columns = list(payload_columns.difference(existing_columns))
+
+        for index, column in enumerate(missing_columns):
+            self.runtime.logger.info("Missing column {}. {}".format(index + 1, column))
+            return_columns.append((column, self.column_identifier.identify(attr_payload[column])))
+        return return_columns
+
+    def handle_missing_table(self, table_name):
+
+        table_exist_status = self.check_table_exist(table_name)
+
+        if not table_exist_status:
+            self.runtime.logger.info("Table [{}] not found in database [{}]."
+                                     .format(table_name, os.environ[constants.DB_NAME]))
+            cursor = self.connection.cursor()
+            cursor.execute(f'create table {table_name}(log_{table_name}_id bigint auto_increment primary key)')
+            cursor.close()
+            self._table_column_cache[table_name] = dict()
+            self._table_column_cache[table_name]["log_" + table_name + "_id"] = True
+            self.runtime.logger.info("Successfully create table [{}] in database [{}]."
+                                     .format(table_name, os.environ[constants.DB_NAME]))
+
+    def handle_missing_columns(self, payload, table_name):
+
+        missing_columns = self.get_missing_columns(payload, table_name)
+
+        if missing_columns:
+            cursor = self.connection.cursor()
+
+            for column in missing_columns:
+                missing_column = column[0]
+                column_type = column[1][0]
+                cursor.execute(f"alter table {table_name} add column `{missing_column}` {column_type}")
+                self._table_column_cache[table_name][missing_column] = True
+
+                self.runtime.logger.info("Added new column [{}] of identified type [{}] to table [{}] "
+                                         "of database [{}].".format(missing_column, column_type,
+                                                                    table_name, os.environ[constants.DB_NAME]))
+
+            cursor.close()
+
+    def insert_payload(self, payload, table_name, dry_run=False):
+
+        try:
+            columns, values = [], []
+
+            for column in payload:
+                columns.append(f'`{column}`')
+                values.append(f'"{payload[column]}"')
+
+            columns = ",".join(columns)
+            values = ",".join(values)
+
+            insert_string = f"insert into {table_name}({columns}) values({values})"
+
+            if not dry_run:
+                cursor = self.connection.cursor()
+                cursor.execute(insert_string)
+                self.connection.commit()
+                cursor.close()
+                return 0
+            else:
+                print(insert_string)
+                return 0
+        except:
```

Comment:
despite the many existing bad examples, [we really should not have a bare except](https://moserei.de/2017/09/20/python-bare-except.html). I get really annoyed when I press `^C` (or hit "cancel" in jenkins) and it is ignored.
```suggestion
        except Exception:
```